plugin/lib/class-php-compatibility-checker.php

```diff
@@ -109,66 +108,104 @@ public function init() {
 	}
 
 	/**
-	 * Generates a list of directories to scan and populate the queue.
+	 * Check if the plugin should be excluded from the test
 	 *
-	 * @since  1.0.0
+	 * @param array $plugin_data Plugin data.
+	 * @see https://developer.wordpress.org/reference/functions/get_plugin_data/
+	 * @return boolean
 	 */
-	public function generate_directory_list() {
-		if ( ! function_exists( 'get_plugins' ) ) {
+	public function exclude_plugin( $plugin_data ) {
+		$excluded_plugins = array( 'PHP Compatibility Checker', 'Hello Dolly' );
```

Comment:
Do we want to ensure these plugins are **always** excluded? Not sure how important that is for these two but I could imagine this list growing. If so, we'll want to move this to be after the filter.

Reply:
The plugin originally excluded these two, probably to keep resources. Hello Dolly is never used example and self plugin assumed always green.

For Hello Dolly, WPTide has a report, so potentially, site owner could scan it and receive results.

Not sure if we ever need the exclusion logic in this new version of the plugin.

I was also thinking to remove this filter at all in favour of `phpcompat_plugins_to_scan`